### PR TITLE
Awesome mixin modification

### PR DESCRIPTION
**Pull request by bandrzejczak**
TimePrinter class was a decorator, which was nice, but Scala
allows us to create stackable modifications with trait mixins,
so frankly, I just wanted to check it out ;)

Sue me ;)

### CODE REVIEW

File: src/main/scala/pl/com/bms/fileSynchronizer/fileUploader/TimePrinter.scala [hunk 1]
```diff
@@ -3,12 +3,12 @@ package pl.com.bms.fileSynchronizer.fileUploader
 import java.text.SimpleDateFormat
 import java.util.Date
 
-class TimePrinter(val fileUploader: FileUploader) extends FileUploader{
-  
+trait TimePrinter extends FileUploader {
+
   val dateFormat = new SimpleDateFormat("HH:mm:ss")
```
Comment (reverendus) [inline src/main/scala/pl/com/bms/fileSynchronizer/fileUploader/TimePrinter.scala]: This isn't thread-safe actually, and although it's quite likely that nothing wrong will happen (as there is no multi-threading in the app), I would suggest fixing that anyway.

Comment (bandrzejczak) [inline src/main/scala/pl/com/bms/fileSynchronizer/fileUploader/TimePrinter.scala]: Is there a better way to do it, then just to inline SimpleDateFormat?

Comment (reverendus) [inline src/main/scala/pl/com/bms/fileSynchronizer/fileUploader/TimePrinter.scala]: Hmm... nothing better I can think of :/

**Comment by wojda** [comment]
Finally we started writing in Scala-way. Great.

**Pull request merged**